org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/diagnostics/DiagnosticsResult.java

```diff
@@ -0,0 +1,28 @@
+package org.eclipse.lemminx.services.extensions.diagnostics;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.lsp4j.Diagnostic;
+
+public class DiagnosticsResult extends ArrayList<Diagnostic> {
+
+	private static final long serialVersionUID = 1L;
+
+	public static final DiagnosticsResult EMPTY = new DiagnosticsResult();

```

Comment:
beware, EMPTY is mutable (can call addFutures)

Reply:
fixed